commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPOptionCategoryLocalServiceImpl.java

```diff
@@ -63,10 +64,16 @@
 	@Override
 	public CPOptionCategory addCPOptionCategory(
 			long userId, Map<Locale, String> titleMap,
-			Map<Locale, String> descriptionMap, double priority, String key,
-			ServiceContext serviceContext)
+			Map<Locale, String> descriptionMap, double priority, String key)
 		throws PortalException {
 
+		ServiceContext serviceContext =
```

Comment:
I don't like Service context, but I do like fewer ThreadLocals :) 

In this case, if it is mandatory to have, then better pass via parameter.

What do you think?

Reply:
Big problem is that Zoltan provided ServiceContextHelperImpl.getServiceContext() in headless-commerce-core-impl which builds service context using NULLs for references and ZEROs for longs (like group ID). IT IS CRAZY ANTI PATTERN and you guys let it IN. Then Alessio took that helper and had used that crazy helper's method and used it around headless impl... let me roll-back this, and check how I can fix the mess. Unfortunately day has only 24 hours. You really putt me on fire with this.

Reply:
I am asking you just to leave the service context as parameter, we are already solvin th eservicecontext helper implementation.

Thanks